axis/door_control.py

```diff
@@ -0,0 +1,188 @@
+
+SUPPORTED_CAPABILITIES = (
+    CAPABILITY_ACCESS,
+    CAPABILITY_LOCK,
+    CAPABILITY_UNLOCK,
+    CAPABILITY_BLOCK,
+    CAPABILITY_DOUBLE_LOCK,
+    CAPABILITY_LOCK_DOWN,
+    CAPABILITY_LOCK_OPEN,
+    CAPABILITY_DOOR_MONITOR,
+    CAPABILITY_LOCK_MONITOR,
+    CAPABILITY_DOUBLE_LOCK_MONITOR,
+    CAPABILITY_ALARM,
+    CAPABILITY_TAMPER,
+    CAPABILITY_WARNING,
+    CAPABILITY_CONFIGURABLE
+)
+
+
+class DoorControl(APIItems):
+    """Door control for Axis devices."""
+
+    def __init__(self, request: object) -> None:
+        super().__init__({}, request, URL, Door)
+
+    # TODO: Question: Is this used to get status information for the door?  Or to update the object properties?
+    async def update(self) -> None:
+        raw = await self.get_door_info_list()
+        self.process_raw(raw)
+
+    @staticmethod
+    def pre_process_raw(raw: dict) -> dict:
+        """Return a dictionary of doors."""
+        door_control_data = raw.get("DoorInfo", {})
+        return {api["token"]: api for api in door_control_data}
+
+    async def get_service_capabilities(self) -> dict:
+        """List the capabilities of the door controller."""
+        return await self._request(
+            "post",
+            URL,
+            json={"tdc:GetServiceCapabilities": {}},
+        )
+
+    async def get_door_info_list(self) -> dict:
+        """List the doors."""
+        return await self._request(
+            "post",
+            URL,
+            json={"tdc:GetDoorInfoList": {}},
+        )
+
+    async def get_door_info(self, door_tokens: list) -> dict:
+        """List the door information."""
+        return await self._request(
+            "post",
+            URL,
+            json={"tdc:GetDoorInfo": {"Token": door_tokens}}
+        )
+
+    async def get_door_state(self, door_token: str) -> dict:
+        """List the door information."""
+        return await self._request(
+            "post",
+            URL,
+            json={"tdc:GetDoorState": {"Token": door_token}}
+        )
+
+    # region Door Actions
+    async def access_door(self, door_token: str) -> None:
+        """Access the door. Use when a credential holder is granted access,
+         for example by swiping a card in a card reader."""
+        await self._request(
+            "post",
+            URL,
+            json={"tdc:AccessDoor": {"Token": door_token}}
+        )
+
+    async def lock_door(self, door_token: str) -> None:
+        """Lock the door."""
+        await self._request(
+            "post",
+            URL,
+            json={"tdc:LockDoor": {"Token": door_token}}
+        )
+
+    async def unlock_door(self, door_token: str) -> None:
+        """Unlock the door until it is explicitly locked again."""
+        await self._request(
+            "post",
+            URL,
+            json={"tdc:UnlockDoor": {"Token": door_token}}
+        )
+
+    async def block_door(self, door_token: str) -> None:
+        """Block the door."""
```

Comment:
What does this mean? Is the door in a specific state so it cant be unlocked? Whats different to this command compared with lock_down_door?

Reply:
I'll add more doc description that reflects the actions and states as described in the Axis API, but some of these features I'm not using in my current set up, so I don't know off the top of my head.

Reply:
Do your best, but if the documentation doesn't help you it can wait til we need to utilize it

Reply:
Docs updated and I used much of the API documentation as it was pretty plain language to each command and how it affects the change in DoorMode.

Reply:
Once branch conflicts are resolved and your code runs against the test suites you're gonna have a swell time with pydocstyle 😂